EKF/gps_yaw_fusion.cpp

```diff
@@ -68,88 +69,62 @@ void Ekf::fuseGpsYaw()
 	// calculate predicted antenna yaw angle
 	const float predicted_hdg =  atan2f(ant_vec_ef(1),ant_vec_ef(0));
 
-	// calculate observation jacobian
-	float t2 = sinf(_gps_yaw_offset);
-	float t3 = cosf(_gps_yaw_offset);
-	float t4 = q0*q3*2.0f;
-	float t5 = q0*q0;
-	float t6 = q1*q1;
-	float t7 = q2*q2;
-	float t8 = q3*q3;
-	float t9 = q1*q2*2.0f;
-	float t10 = t5+t6-t7-t8;
-	float t11 = t3*t10;
-	float t12 = t4+t9;
-	float t13 = t3*t12;
-	float t14 = t5-t6+t7-t8;
-	float t15 = t2*t14;
-	float t16 = t13+t15;
-	float t17 = t4-t9;
-	float t19 = t2*t17;
-	float t20 = t11-t19;
-	float t18 = (t20*t20);
-	if (t18 < 1e-6f) {
+	// using magnetic heading process noise
+	// TODO extend interface to use yaw uncertainty provided by GPS if available
+	const float R_YAW = sq(fmaxf(_params.mag_heading_noise, 1.0e-2f));
+
+	// calculate intermediate variables
+	const float HK0 = sinf(_gps_yaw_offset);
+	const float HK1 = q0*q3;
+	const float HK2 = q1*q2;
+	const float HK3 = 2*HK0*(HK1 - HK2);
+	const float HK4 = cosf(_gps_yaw_offset);
+	const float HK5 = ecl::powf(q1, 2);
```

Comment:
What's the reason for using `ecl::powf(q1, 2)` instead of `q1*q1`?

Reply:
Sympy generates `powf(a,b)` a lot, so we added a custom `ecl::powf` which only accepts integer powers and just does the multiplication so that we don't accidentally do `powf(float,float)` style calculations which the compiler can't optimize to something like `x*x`. Then it's easy to just find/replace powf with ecl::powf and make sure everything is covered.